src/WooCommerce.php

```diff
@@ -386,10 +397,22 @@ public static function woocommerce_save_product_variation($post_id, $loop) {
       '_' . Plugin::PREFIX . '_gtin',
       '_' . Plugin::PREFIX . '_erp_inventory_id',
     ];
+
     foreach ($custom_fields as $field) {
       if (isset($_POST[$field]) && isset($_POST[$field][$loop])) {
         if ($_POST[$field][$loop]) {
-          update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+          if ($field !== '_' . Plugin::PREFIX . '_gtin') {
+            update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+          }
+          else {
+            if (!self::is_existing_gtin($variation_id, $_POST[$field][$loop])) {
+              // GTIN is only saved if unique.
+              update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+            }
+            else {
+              \WC_Admin_Meta_Boxes::add_error(__('The entered GTIN isn\'t unique and couldn\'t be saved. Please enter a unique GTIN and save again. All other changes have been saved.', Plugin::L10N));
```

Comment:
This code looks duplicated to me. Line 235 of this file is doing the same. Can we extract this into a function? Would be easier to manage/change the translation in that way.